src/client/src/App.jsx

```diff
@@ -0,0 +1,79 @@
+// #region imports
+import { useEffect, useMemo, useState } from 'react';
+import './App.css'
+import { MessageForm } from './MessageForm.jsx';
+import { MessageList } from './MessageList.jsx';
+import { apiUtils } from './API/apiUtils.js';
+import { Room } from './Room.jsx';
+// #endregion
+
+const DataLoader = ({ onData, room, userName }) => {
+  const WS_URL = 'ws://localhost:3005';
+  useEffect(() => {
+    const socket = new WebSocket(WS_URL);
+
+    socket.addEventListener('open', () => {
+      socket.send(JSON.stringify(
+        { text: `${userName} join the room`, userName, room, type: 'join' }
+      ));
+    });
+
+    socket.addEventListener('message', (event) => {
+      const text = JSON.parse(event.data)
+
+      onData(text);
+    });
+
+    return () => {
+      socket.close();
+    };
+  }, [room])
+
+  return (
+    <h1 className="title">Chat application</h1>
+  );
+};
+
+export function App() {
+  const [messages, setMessages] = useState([]);
+  const [userName, setUserName] = useState(null);
+  const [room, setRoom] = useState('room1');
+
+  const sortedMessages = useMemo(() => {
+    return messages.filter(message => message.room === room);
+  },[messages, room]);
+
+  useEffect(() => {
+    if (!localStorage.getItem('name')) {
```

Comment:
The user's name is requested every time the 'room' state changes. The dependency array should be empty so that the name is only requested on the first render.